lib/routes/ups/track.ts

```diff
@@ -0,0 +1,107 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+import puppeteer from '@/utils/puppeteer';
+
+export const route: Route = {
+    path: '/track/:trackingNumber',
+    categories: ['other'],
+    example: '/ups/track/1Z78R6790470567520',
+    parameters: { trackingNumber: 'The UPS tracking number (e.g., 1Z78R6790470567520).' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['ups.com/track?loc=en_US&tracknum=:trackingNumber'],
+            target: '/ups/track/:trackingNumber',
+        },
+    ],
+    name: 'UPS Tracking',
+    maintainers: ['Aquabet'],
+    handler,
+};
+
+async function handler(ctx) {
+    const { trackingNumber } = ctx.req.param();
+    const url = `https://www.ups.com/track?loc=en_US&tracknum=${trackingNumber}`;
+
+    const browser = await puppeteer();
+    const page = await browser.newPage();
+
+    await page.setRequestInterception(true);
+
+    // skip loading images, stylesheets, and fonts
+    page.on('request', (request) => {
+        if (['image', 'stylesheet', 'font'].includes(request.resourceType())) {
+            request.abort();
+        } else {
+            request.continue();
+        }
+    });
+
+    await page.goto(url, { waitUntil: 'domcontentloaded' });
+
+    await page.waitForSelector('tr[id^="stApp_ShpmtProg_LVP_progress_row_"]', {
+        timeout: 30000,
```

Comment:
Do not wait for 30s, the browser will be closed by RSSHub https://github.com/DIYgod/RSSHub/blob/f5fc827bb8bda196844330f5907422c5d8d680ec/lib/utils/puppeteer.ts#L57-L59